crates/astria-sequencer-types/src/namespace.rs

```diff
@@ -0,0 +1,134 @@
+use std::{
+    fmt,
+    ops::Deref,
+};
+
+use astria_celestia_jsonrpc_client::blob::NAMESPACE_ID_AVAILABLE_LEN;
+use eyre::{
+    ensure,
+    WrapErr as _,
+};
+use serde::{
+    de::{
+        self,
+        Visitor,
+    },
+    Deserialize,
+    Deserializer,
+    Serialize,
+};
+use sha2::{
+    Digest,
+    Sha256,
+};
+
+/// The default namespace blocks are written to.
+/// A block in this namespace contains "pointers" to the rollup txs contained
+/// in that block; ie. a list of tuples of (DA block height, namespace).
+pub static DEFAULT_NAMESPACE: Namespace = Namespace(*b"astriasequ");
+
+/// Namespace represents a Celestia namespace.
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
+pub struct Namespace([u8; NAMESPACE_ID_AVAILABLE_LEN]);
+
+impl Deref for Namespace {
+    type Target = [u8; NAMESPACE_ID_AVAILABLE_LEN];
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl Namespace {
+    #[must_use]
+    pub fn new(inner: [u8; NAMESPACE_ID_AVAILABLE_LEN]) -> Self {
+        Self(inner)
+    }
+
+    /// Creates a namespace from a 10-byte hex-encoded string.
+    ///
+    /// # Errors
+    ///
+    /// - if the string cannot be deocded as hex
+    /// - if the string does not contain 10 bytes
+    pub fn from_string(s: &str) -> eyre::Result<Self> {
```

Comment:
Remove. This is mainly a representation/serialization concern.

Reply:
changed to private since it's used in this file for deserialization only